service/src/main/kotlin/fi/espoo/evaka/invoicing/service/DraftInvoiceGenerator.kt

```diff
@@ -278,11 +292,31 @@ class DraftInvoiceGenerator(
                                                     part.placement.unitId,
                                                     part.placement.type
                                                 ),
-                                                part.fee,
+                                                when {
+                                                    isHalfFeeMonth(part) ->
+                                                        calculateHalfFee(part.fee)
+                                                    else -> part.fee
+                                                },
                                                 part.feeAlterations.map { feeAlteration ->
                                                     Pair(feeAlteration.type, feeAlteration.effect)
                                                 },
-                                                part.finalFee,
+                                                when {
+                                                    isHalfFeeMonth(part) -> {
+                                                        val fee = calculateHalfFee(part.fee)
+                                                        part.feeAlterations.fold(fee) {
+                                                            currentFee,
+                                                            feeAlteration ->
+                                                            currentFee +
+                                                                feeAlterationEffect(
+                                                                    currentFee,
+                                                                    feeAlteration.type,
+                                                                    feeAlteration.amount,
+                                                                    feeAlteration.isAbsolute
+                                                                )
+                                                        }
+                                                    }
```

Comment:
I don't understand why fee alterations are summed here, and there seems to be no test for this case

Reply:
`finalFee = fee + feeAlterations` and if we just use `calculateHalfFee(part.finalFee)` here and absolute fee alterations are used, alterations are also halved which is probably wrong.

Reply:
Oh and I added a test.